src/main/java/at/ac/tuwien/kr/alpha/Alpha.java

```diff
@@ -0,0 +1,157 @@
+package at.ac.tuwien.kr.alpha;
+
+import at.ac.tuwien.kr.alpha.common.AnswerSet;
+import at.ac.tuwien.kr.alpha.common.Program;
+import at.ac.tuwien.kr.alpha.common.atoms.Atom;
+import at.ac.tuwien.kr.alpha.common.atoms.BasicAtom;
+import at.ac.tuwien.kr.alpha.common.predicates.*;
+import at.ac.tuwien.kr.alpha.common.terms.ConstantTerm;
+import at.ac.tuwien.kr.alpha.grounder.Grounder;
+import at.ac.tuwien.kr.alpha.grounder.GrounderFactory;
+import at.ac.tuwien.kr.alpha.grounder.parser.ProgramParser;
+import at.ac.tuwien.kr.alpha.solver.Solver;
+import at.ac.tuwien.kr.alpha.solver.SolverFactory;
+import at.ac.tuwien.kr.alpha.solver.heuristics.BranchingHeuristicFactory;
+import org.reflections.Reflections;
+import org.reflections.scanners.MethodAnnotationsScanner;
+import org.reflections.util.ClasspathHelper;
+import org.reflections.util.ConfigurationBuilder;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.util.*;
+import java.util.stream.Stream;
+
+public class Alpha {
+	private final Map<String, FixedInterpretationPredicate> predicateMethods = new HashMap<>();
+
+	private final String grounderName;
+	private final String solverName;
+	private final String storeName;
+
+	private Solver solver;
+
+	private Program program;
+
+	public long getSeed() {
+		return seed;
+	}
+
+	public void setSeed(long seed) {
+		this.seed = seed;
+	}
+
+	public void setDeterministic() {
+		setSeed(0);
+	}
+
+	private long seed;
+
+	public Alpha(String grounderName, String solverName, String storeName) {
+		this.grounderName = grounderName;
+		this.solverName = solverName;
+		this.storeName = storeName;
+	}
+
+	public Alpha(String grounderName, String solverName) {
+		this(grounderName, solverName, "alpharoaming");
+	}
+
+	public Alpha(String grounderName) {
+		this(grounderName, "default");
+	}
+
+	public Alpha() {
+		this("naive");
+	}
+
```

Comment:
Yes, this will eventually evolve into the outwards facing API, but I expect more changes to follow, so I would prefer to get this branch merged before, and iterate on it afterwards. We can always refine it.

I already have a handful of other changes that depend on this PR, including code that better showcases how such an "outwards facing" API might be used.

Reply:
Ok sounds reasonable to refine it later when things have settled. I also have some changes looking forward to this major PR to come into effect.